bin/onlyswaps-state-api/src/network_bus.rs

```diff
@@ -0,0 +1,70 @@
+use alloy::network::EthereumWallet;
+use alloy::primitives::{Address, FixedBytes};
+use alloy::providers::{DynProvider, Provider, ProviderBuilder, WsConnect};
+use alloy::signers::local::PrivateKeySigner;
+use config::network::NetworkConfig;
+use generated::onlyswaps::router::IRouter::SwapRequestParameters;
+use generated::onlyswaps::router::Router::{RouterInstance, getSwapRequestReceiptReturn};
+use std::collections::HashMap;
+
+pub(crate) struct NetworkBus<P> {
+    pub networks: HashMap<u64, Network<P>>,
+}
+
+impl NetworkBus<DynProvider> {
+    pub async fn create(network_configs: &[NetworkConfig]) -> anyhow::Result<Self> {
+        let mut networks = HashMap::new();
+
+        for config in network_configs.iter() {
+            let network = Network::new(config).await?;
+            networks.insert(config.chain_id, network);
+        }
+
+        Ok(Self { networks })
+    }
+}
+
+pub(crate) struct Network<P> {
```

Comment:
this is a sort of anaemic struct - it's basically a create_provider with some extra functions added from the contract instance